docs/eventModelling.md

```diff
@@ -0,0 +1,48 @@
+User saves a draft email.|```<Create>``` with an ```<Object>``` of type ```<Email>```
+User clicks on "Export results to XLS" within an application.|```<Export>```with an ```<Object>``` of type ```<File>```|This is an ```<Export>``` as it relates to information being transferred out of a controlled area (the application).
+User clicks on "Print to PDF" within an application.|```<Export>```with an ```<Object>``` of type ```<File>```|N.B. not ```<Print>```, as it doesn't relate to creating hard-copy, but it is an ```<Export>``` as it relates to information being transferred out of a controlled area (the application).
+Web proxy allows an authenticated user access to a website of known category.|```<View>``` of type ```<Resource>``` that has ```<Category>``` element assigned to that reported by the web proxy|Events are modelled from the point of view of the user, where possible.
+Someone plugs a USB drive into a workstation while a user session is logged in.|```<Install>``` of type ```<Media>```.  ```<User>``` is the user currently logged into the workstation.|
+Someone unplugs a USB drive into a workstation while a user session is not logged in.|```<Uninstall>``` of type ```<Media>```.  There is no  ```<User>```  element.|
+Antivirus product on a workstation with a logged in user detects malware on a file.|```<AntiMalware>``` with sub-element ```<Quarantine>``` |*This will be modelled as ```<Alert>``` in a future version of schema.*
+An IPS appliance blocks an IP address that may be attempting an SQL injection attack.|```<Alert>``` with sub-element ```<IDS>```.  Attacker IP is recorded in ```<Source>``` and target in ```<Destination>```.|
+A database reports that a processing account logs on and performs a backup.|```<Authenticate>``` and then a second event, ideally ```<Process>``` if the backup is contained within the database and control maintained, otherwise ```<Export>```.| It might be unclear that this is a backup operation, e.g. perhaps a query is performed.  In this case ```<Search>``` may be most appropriate for the second event.
+User sends a document to the print queue for immediate printing.|```<Print>``` with ```<Action>``` set to ```CreateJob```|Q: Why not ```StartPrint```? A: Although user intends this to be printed immediately, it might actually be delayed or never actually be printed, e.g. due to an error.
+Print server or network printer reports that it has started printing a document.|```<Print>``` with ```<Action>``` set to ```StartPrint```.|
+Administrator creates a new user.|```<Create>``` of type ```<User>```|
+Administrator creates a new role for power-users.|```<Create>``` of type ```<Group>```|
+Administrator assigns a user to the power-users role.|```<Authorise>``` of type ```<User>``` with ```<AddGroups>``` element assigned appropriately.|
```

Comment:
I wonder if this sort of stuff should be dealt with using Create/Update/Delete/View with entity types of User/Group/Role/etc. Athorise element seems all wrong, but this is a major change.